subgraphs/saddle-finance/src/entities/pool.ts

```diff
@@ -422,29 +431,117 @@ function updateRewardTokenEmissionsUSD(
   pool.rewardTokenEmissionsUSD = rewardTokenEmissionsUSD;
 }
 
+// addLPVolume will convert a given LP amount from a swap into the underlying equivalent
+// and add it to an array of pool inputTokenVolumes. If the swap doesn't involve the LP
+// it won't do anything.
+function addLPVolume(
+  pool: LiquidityPool,
+  swap: SwapEvent,
+  poolVolumes: BigInt[]
+): BigInt[] {
+  const basePool = LiquidityPool.load(pool._basePool!)!;
+  if (basePool.outputTokenSupply!.equals(BIGINT_ZERO)) {
+    return poolVolumes;
+  }
+
+  if (
+    swap.tokenIn != basePool.outputToken &&
+    swap.tokenOut != basePool.outputToken
+  ) {
+    return poolVolumes;
+  }
+
+  let lpAmount = swap.amountIn;
+  if (swap.tokenOut == basePool.outputToken) {
+    lpAmount = swap.amountOut;
+  }
+
+  const multiplier = lpAmount.divDecimal(
+    basePool.outputTokenSupply!.toBigDecimal()
+  );
+  for (let i = 0; i < basePool.inputTokens.length; i++) {
+    const token = basePool.inputTokens[i];
+    const balance = basePool.inputTokenBalances[i].toBigDecimal();
+    const tokenIndex = pool.inputTokens.indexOf(token);
+
+    const vol = bigDecimalToBigInt(balance.times(multiplier));
+    poolVolumes[tokenIndex] = poolVolumes[tokenIndex].plus(vol);
+  }
+  return poolVolumes;
+}
+
 function addTokenVolume(
   tokenVolume: BigInt[],
   swap: SwapEvent,
   pool: LiquidityPool
 ): BigInt[] {
+  if (pool._basePool) {
+    tokenVolume = addLPVolume(pool, swap, tokenVolume);
```

Comment:
It shouldn't because the volume is added in terms of both input and output tokens. The LP will only be one side (input/output), and we still need to add the volume of the other one. But I can see how it can be confusing, let me see if I can reorg it in a clearer way